### ISSUE TITLE
Blinds only UP/DOWN no STOP option

### ISSUE THREAD

**Issue by JHurk** [PatchworkBoy/homebridge-edomoticz#247]
I know this one is old and there are more issues that seem similar to this one, but I just installed blinds and found myself searching for the 'stop' button.
I find some issues that mention this a part of the issue or that the stop is a new feature in Domoticz.
But have not been able to find a solution, hopefully you can help me.

**Problem;**
I have two MOES  MS-108ZR installed on my Zigbee network for two shutters (rolling)
Zigbee2MQTT --> MQTT --> Domoticz (switch type=blinds percentage) --> Homebridge --> Homekit
The 2 shutter work perfect when I control then with Zigbee2MQTT UI (Up/Down/Stop and percentage slider)
But when I control them in Domoticz (and Homebridge and Homekit) there is no Stop button.
I can change the switch type in Domoticz to 'blinds + stop' and then a Stop button appears in Domoticz.
In Domoticz this works great, but Homekit tells me that 'switch type is not recognized' and only shows the switch with this message and no buttons or sliders to operate them.

**Solution/question;**
Is this due to the plugin homebridge-edomoticz? Does someone need to add this new type of switch ('blinds + stop') to the plugin, or does Homekit prevent this from working (does Homekit cause this issue)?
I seem to have seen some switches in Homekit that have a stop button, but am unable to get this confirmed so maybe I was dreaming this :).

**Comment by PatchworkBoy** [comment]
Question is whether HomeKit supports blinds with a stop option… needs to be a suitable Service type for the ‘entire’ device with the relevant Characteristics available for each control element of that device within the HomeKit specs at https://github.com/homebridge/HAP-NodeJS/tree/master/src/lib/definitions

if there’s no native support within HomeKit for the combination of controls required within the Service, then it’s only doable as a custom device type, which then loses a lot of the Siri / automation functionality etc.

simplest way would be create a virtual switch in Domoticz mapped to the working stop control. This’ll appear in HomeKit as a plain on/off switch. In Domoticz give it an auto-off timeout and set it as a push button.

Call the button Blind Stop. Drop it into a room in HomeKit, then say “Hey Siri, [roomname] Blind Stop”

**Comment by PatchworkBoy** [comment]
(Blinds are the bane of my existence, as I don’t own any with which to test / experiment… and it would appear the variation between requirements of non-native blinds is MASSIVE)

**Comment by JHurk** [comment]
@PatchworkBoy Thanks for the reply and suggestion. Like I said; I do not know if this is supported by Homekit.

I tried your suggestion by making a virtual switch with the stop command (and auto-off script to make I a 'push button' in Homekit. This works great!
Command I used in the 'On action' field in the virtual switch;
http://<ip of domoticz>:<domoticz port, 8084 in my case>/json.htm?type=command&param=switchlight&idx=<IDX of the shutter switch in Domoticz>&switchcmd=Stop

Now I have ran into a new problem, maybe you know how to fix this?
One of the 2 blinds keeps reporting in Homebridge that it is open 18%, while it is fully closed.
I can control the blind within Homekit, Homebidge, Domoticz and Zigbee2Mqtt but after the operation is finished (going up or down) Homebridge and Homekit report is as 18% open. While Domoticz and Zigbee2Mqtt give the actual state that the shutter is in.

I have deleted the accessory in Homebridge and restarted Homebridge and adding it again, but is keeps reporting the 18%.
I have tried setting the slider from 18% to another state. This works; the shutter moves, but after the operation Homebridge and Homekit report the shutter as 18% open, while Domticz and Zigbee2Mqtt report the correct status.

Any suggestion on how to fix this?
